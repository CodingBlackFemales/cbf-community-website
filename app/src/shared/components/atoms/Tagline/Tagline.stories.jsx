import React from 'react';

import { storiesOf } from '@storybook/react';

import Tagline from '.';

storiesOf('Atoms', module).add('Tagline', () => (
  <Tagline text={'A tagline | with some text'} />
));
