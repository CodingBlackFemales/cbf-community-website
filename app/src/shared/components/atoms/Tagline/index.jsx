import React from 'react';
import styled from 'styled-components';

const Tagline = ({ text, className }) => {
  const classPrefix = 'tagline';
  return (
    <div className={(classPrefix, className)}>
      <p className={`${classPrefix}__text`}>{text}</p>
    </div>
  );
};

const StyledTagline = styled(Tagline)`
  text-align: center;

  .tagline__text {
    margin-top: 0.2em;
    font-weight: 400;
  }
`;

export default StyledTagline;
