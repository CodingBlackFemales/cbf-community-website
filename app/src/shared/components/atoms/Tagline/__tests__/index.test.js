import React from 'react';
import { cleanup } from '@testing-library/react';

import Tagline from '..';
import {
  mountWithTheme,
  renderWithTheme,
} from '../../../../utils/test/testHelpers';

describe('<Tagline', () => {
  afterEach(cleanup);
  const taglineComponent = <Tagline text={'A tagline'} />;

  it('matches the snapshot, when the label and tagline props exist', () => {
    const { asFragment } = renderWithTheme(taglineComponent);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the tagline text', () => {
    const component = mountWithTheme(taglineComponent);

    const taglineText = component.find('.tagline__text');

    expect(taglineText.exists()).toBe(true);
  });
});
