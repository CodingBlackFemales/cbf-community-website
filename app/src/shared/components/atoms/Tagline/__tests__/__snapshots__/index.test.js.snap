// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tagline matches the snapshot, when the label and tagline props exist 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
}

.c0 .tagline__text {
  margin-top: 0.2em;
  font-weight: 400;
}

<div
    class="c0"
  >
    <p
      class="tagline__text"
    >
      A tagline
    </p>
  </div>
</DocumentFragment>
`;
